corehq/motech/models.py

```diff
@@ -163,6 +168,85 @@ def last_token(self, token: Optional[dict]):
     def notify_addresses(self):
         return [addr for addr in re.split('[, ]+', self.notify_addresses_str) if addr]
 
+    @property
+    def is_throttled_connection(self):
```

Comment:
This adds a lot of bloat to the connection settings model. What about extracting it to a separate class? That would also make it easier to swap out in future and possibly easier to test.

Reply:
I'm glad you mention this, cause I couldn't make up my mind on moving this out.